### DISCUSSION TITLE
Manhattan router keeps jumping

### DISCUSSION THREAD

**Discussion by seekeramento** [clientIO/joint#1928] [Q&A]
I am using Manhattan as my defaultRouter. 

```js
defaultRouter: function(vertices, opt, linkView) {
      const manhattanVertices = joint.routers.manhattan.call(this, vertices, {
        step: 10,
        perpendicular: false,
        precision: 1
      }, linkView);
      return manhattanVertices;
},
```

It seems to work OK, but when I move a connector around, I can see a lot of jumping happening around.
![routingIssue](https://user-images.githubusercontent.com/99613862/205563960-4527e4f4-dd15-40cf-9ef3-e6756eb03d38.gif)

What options should I use in order to overcome this issue?

**Comment by kumilingus** [comment]
The flickering is a known issue of the `manhattan` router. You can try to minimalize it by:
- Setting the paper `gridSize` to the same value as `manhattan` `step` (i.e. `10`).
- The `connectionPoints` must lie in the grid, for that the height of each element should a multiple of `2 * gridSize`. Alternatively, an `anchor` has to be set, to make sure of the same.

**Reply by seekeramento** [reply]
Thanks for the clarification. Do you have a relevant code example?

**Reply by kumilingus** [reply]
```js
paper.options.gridSize = 10;
el.resize(5 * gridSize, 2 * gridSize * 10);
```

**Reply by kumilingus** [reply]
e.g. the [ChatBot](https://resources.jointjs.com/demos/chatbot) application is written this way.
